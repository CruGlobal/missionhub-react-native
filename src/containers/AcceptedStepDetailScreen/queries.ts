import gql from 'graphql-tag';

import { REMINDER_BUTTON_FRAGMENT } from '../../components/ReminderButton/queries';
import { REMINDER_DATE_TEXT_FRAGMENT } from '../../components/ReminderDateText/queries';
import { STEP_DETAIL_POST_FRAGMENT } from '../../components/StepDetailScreen/queries';

export const ACCEPTED_STEP_DETAIL_QUERY = gql`
  query AcceptedStepDetail($id: ID!) {
    step(id: $id) {
      id
      title
      stepType
      stepSuggestion {
        id
        descriptionMarkdown
      }
      receiver {
        id
        firstName
      }
      community {
        id
      }
      reminder {
        ...ReminderButton
        ...ReminderDateText
      }
      post {
        ...StepDetailPost
      }
    }
  }
  ${REMINDER_BUTTON_FRAGMENT}
  ${REMINDER_DATE_TEXT_FRAGMENT}
  ${STEP_DETAIL_POST_FRAGMENT}
`;

export const DELETE_STEP_MUTATION = gql`
  mutation DeleteStep($input: DeleteStepInput!) {
    deleteStep(input: $input) {
      id
    }
  }
`;

export const DELETE_STEP_REMINDER_MUTATION = gql`
  mutation DeleteReminder($input: DeleteReminderInput!) {
    deleteReminder(input: $input) {
      id
    }
  }
`;
