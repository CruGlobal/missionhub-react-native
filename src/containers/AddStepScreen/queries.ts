import gql from 'graphql-tag';

import { STEP_ADDED_ANALYTICS_FRAGMENT } from '../../actions/analyticsQueries';

export const CREATE_CUSTOM_STEP_MUTATION = gql`
  mutation CreateCustomStep(
    $title: String!
    $receiverId: ID!
    $stepType: StepTypeEnum
  ) {
    createCustomStep(
      input: { title: $title, receiverId: $receiverId, stepType: $stepType }
    ) {
      step {
        ...StepAddedAnalytics
      }
    }
  }
  ${STEP_ADDED_ANALYTICS_FRAGMENT}
`;
