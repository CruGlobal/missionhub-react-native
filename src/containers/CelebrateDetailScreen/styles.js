import { StyleSheet } from 'react-native';

import theme from '../../theme';

export default StyleSheet.create({
  container: {
    backgroundColor: theme.white,
    flex: 1,
    justifyContent: 'center',
  },
  cardStyle: {
    shadowOpacity: 0,
    shadowRadius: 0,
    marginHorizontal: 3,
  },
  backButtonStyle: { color: theme.black },
  headerStyle: { backgroundColor: theme.white },
  rightHeaderItemStyle: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
});
