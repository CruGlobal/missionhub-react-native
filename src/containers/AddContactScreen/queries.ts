import gql from 'graphql-tag';

import { PERSON_FRAGMENT } from '../PersonItem/queries';

export const GET_PERSON = gql`
  query GetPerson($id: ID!) {
    person(id: $id) {
      ...PersonFragment
    }
  }
  ${PERSON_FRAGMENT}
`;
