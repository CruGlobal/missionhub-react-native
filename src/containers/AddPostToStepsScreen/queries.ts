import gql from 'graphql-tag';

import { STEP_DETAIL_POST_FRAGMENT } from '../../components/StepDetailScreen/queries';

export const ADD_POST_TO_MY_STEPS_SCREEN_DETAILS_QUERY = gql`
  query AddPostToMyStepsScreenDetails($feedItemId: ID!) {
    feedItem(id: $feedItemId) {
      id
      createdAt
      subject {
        ... on Post {
          id
          postType
          stepStatus
          ...StepDetailPost
        }
      }
      subjectPerson {
        id
        firstName
        lastName
      }
    }
  }
  ${STEP_DETAIL_POST_FRAGMENT}
`;

export const ADD_POST_TO_MY_STEPS = gql`
  mutation AddPostToMySteps($input: AddPostToMyStepsInput!) {
    addPostToMySteps(input: $input) {
      step {
        id
        title
        post {
          id
          stepStatus
        }
      }
    }
  }
`;
