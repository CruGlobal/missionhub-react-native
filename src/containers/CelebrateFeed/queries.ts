import gql from 'graphql-tag';

import { CELEBRATE_ITEM_FRAGMENT } from '../../components/CelebrateItem/queries';

export const GET_CELEBRATE_FEED = gql`
  query GetCelebrateFeed(
    $communityId: ID!
    $personIds: [ID!] = null
    $hasUnreadComments: Boolean = false
    $celebrateCursor: String
  ) {
    community(id: $communityId) {
      celebrationItems(
        sortBy: createdAt_DESC
        first: 25
        after: $celebrateCursor
        subjectPersonIds: $personIds
        hasUnreadComments: $hasUnreadComments
      ) {
        nodes {
          ...CelebrateItem
        }
        pageInfo {
          endCursor
          hasNextPage
        }
      }
    }
  }
  ${CELEBRATE_ITEM_FRAGMENT}
`;

export const GET_GLOBAL_CELEBRATE_FEED = gql`
  query GetGlobalCelebrateFeed($celebrateCursor: String) {
    globalCommunity {
      celebrationItems(
        sortBy: createdAt_DESC
        first: 25
        after: $celebrateCursor
      ) {
        nodes {
          ...CelebrateItem
        }
        pageInfo {
          endCursor
          hasNextPage
        }
      }
    }
  }
  ${CELEBRATE_ITEM_FRAGMENT}
`;
