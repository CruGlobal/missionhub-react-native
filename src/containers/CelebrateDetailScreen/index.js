import React, { Component } from 'react';
import { View, SafeAreaView, StatusBar } from 'react-native';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
// eslint-disable-next-line import/default
import ParallaxScrollView from 'react-native-parallax-scroll-view';

import CommentLikeComponent from '../CommentLikeComponent';
import { celebrationItemSelector } from '../../selectors/celebration';
import CelebrateItem from '../../components/CelebrateItem';
import CommentsList from '../CommentsList';
import BackButton from '../BackButton';
import CelebrateCommentBox from '../../components/CelebrateCommentBox';
import theme from '../../theme';
import Header from '../../containers/Header';
import ItemHeaderText from '../../components/ItemHeaderText';
import { PARALLAX_HEADER_HEIGHT } from '../../constants';

import styles from './styles';

class CelebrateDetailScreen extends Component {
  renderBackButton = () => {
    const { backButtonStyle } = styles;

    return <BackButton iconStyle={backButtonStyle} customIcon="deleteIcon" />;
  };

  renderForeground = () => {
    const { event } = this.props;
    const { cardStyle } = styles;

    return (
      <CelebrateItem
        event={event}
        cardStyle={cardStyle}
        rightCorner={this.renderBackButton()}
        namePressable={true}
      />
    );
  };

  renderStickyHeader = () => {
    const { event } = this.props;
    const { subject_person_name } = event;
    const { headerStyle, leftHeaderItemStyle, rightHeaderItemStyle } = styles;

    return (
      <Header
        shadow={false}
        left={
          <ItemHeaderText
            text={subject_person_name}
            style={leftHeaderItemStyle}
          />
        }
        right={
          <View style={rightHeaderItemStyle}>
            <CommentLikeComponent event={event} />
            {this.renderBackButton()}
          </View>
        }
        style={headerStyle}
      />
    );
  };

  render() {
    const { event } = this.props;
    const { container } = styles;

    return (
      <SafeAreaView style={container}>
        <StatusBar {...theme.statusBar.darkContent} />
        <ParallaxScrollView
          backgroundColor={theme.white}
          parallaxHeaderHeight={PARALLAX_HEADER_HEIGHT}
          renderForeground={this.renderForeground}
          stickyHeaderHeight={theme.headerHeight}
          renderStickyHeader={this.renderStickyHeader}
        >
          <CommentsList event={event} organizationId={event.organization.id} />
        </ParallaxScrollView>
        <CelebrateCommentBox event={event} />
      </SafeAreaView>
    );
  }
}

CelebrateDetailScreen.propTypes = {
  event: PropTypes.object.isRequired,
};

const mapStateToProps = (
  { organizations },
  {
    navigation: {
      state: {
        params: { event },
      },
    },
  },
) => ({
  event:
    celebrationItemSelector(
      { organizations },
      { eventId: event.id, organizationId: event.organization.id },
    ) || event,
});
export default connect(mapStateToProps)(CelebrateDetailScreen);
export const CELEBRATE_DETAIL_SCREEN = 'nav/CELEBRATE_DETAIL_SCREEN';
