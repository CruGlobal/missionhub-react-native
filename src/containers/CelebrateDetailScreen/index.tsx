import React, { useRef } from 'react';
import { Image, View, SafeAreaView, StatusBar, FlatList } from 'react-native';
import { connect } from 'react-redux-legacy';
import { AnyAction } from 'redux';
import { ThunkDispatch } from 'redux-thunk';
import { useNavigationParam } from 'react-navigation-hooks';
import { useApolloClient } from '@apollo/react-hooks';

import { CommentLikeComponent } from '../../components/CommentLikeComponent';
import { organizationSelector } from '../../selectors/organizations';
import CommentsList from '../CommentsList';
import DeprecatedBackButton from '../DeprecatedBackButton';
import CelebrateCommentBox from '../../components/CelebrateCommentBox';
import theme from '../../theme';
import TRAILS1 from '../../../assets/images/Trailss.png';
import TRAILS2 from '../../../assets/images/TrailGrey.png';
import { reloadCelebrateComments } from '../../actions/celebrateComments';
import { TrackStateContext } from '../../actions/analytics';
import { celebrateCommentsSelector } from '../../selectors/celebrateComments';
import CardTime from '../../components/CardTime';
import { CommunityFeedItemName } from '../../components/CommunityFeedItemName';
import { CommunityFeedItemContent } from '../../components/CommunityFeedItemContent';
import { RefreshControl } from '../../components/common';
import {
  ANALYTICS_ASSIGNMENT_TYPE,
  ANALYTICS_PERMISSION_TYPE,
} from '../../constants';
import { CommunityFeedItem } from '../../components/CommunityFeedItem/__generated__/CommunityFeedItem';
import { COMMUNITY_FEED_ITEM_FRAGMENT } from '../../components/CommunityFeedItem/queries';
import { Organization, OrganizationsState } from '../../reducers/organizations';
import {
  CelebrateCommentsState,
  CelebrateComment,
} from '../../reducers/celebrateComments';
import { AuthState } from '../../reducers/auth';
import {
  getAnalyticsAssignmentType,
  getAnalyticsPermissionType,
} from '../../utils/analytics';
import { useKeyboardListeners } from '../../utils/hooks/useKeyboardListeners';
import { useRefreshing } from '../../utils/hooks/useRefreshing';
import { useAnalytics } from '../../utils/hooks/useAnalytics';
import { navigateBack } from '../../actions/navigation';

import styles from './styles';

export interface CelebrateDetailScreenProps {
  dispatch: ThunkDispatch<{}, {}, AnyAction>;
  organization: Organization;
  /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
  celebrateComments?: { comments: CelebrateComment[]; pagination: any };
  editingCommentId: string | null;
  analyticsAssignmentType: TrackStateContext[typeof ANALYTICS_ASSIGNMENT_TYPE];
  analyticsPermissionType: TrackStateContext[typeof ANALYTICS_PERMISSION_TYPE];
}

const CelebrateDetailScreen = ({
  dispatch,
  organization,
  celebrateComments,
  editingCommentId,
  analyticsAssignmentType,
  analyticsPermissionType,
}: CelebrateDetailScreenProps) => {
  useAnalytics(['celebrate item', 'comments'], {
    screenContext: {
      [ANALYTICS_ASSIGNMENT_TYPE]: analyticsAssignmentType,
      [ANALYTICS_PERMISSION_TYPE]: analyticsPermissionType,
    },
  });
  const client = useApolloClient();
  const navParamsEvent: CommunityFeedItem = useNavigationParam('item');
  const event =
    client.readFragment<CommunityFeedItem>({
      id: `FeedItem:${navParamsEvent.id}`,
      fragment: COMMUNITY_FEED_ITEM_FRAGMENT,
      fragmentName: 'CommunityFeedItem',
    }) || navParamsEvent;

  const eventIsNotLoaded =
    Object.keys(event).length === 1 && Object.keys(event)[0] === 'id';
  if (eventIsNotLoaded) {
    dispatch(navigateBack()); // Would be better to load celebrate item from GraphQL using cache-and-network fetchPolicy but no root query currently exists. Hopefully we can do this when moving to feedItems.
  }

  const onRefreshCelebrateItem: () => void = useNavigationParam(
    'onRefreshCelebrateItem',
  );

  const listRef = useRef<FlatList<CelebrateComment>>(null);

  const scrollToEnd = () => listRef.current && listRef.current.scrollToEnd();

  const scrollToFocusedRef = () => {
    if (editingCommentId) {
      const index =
        celebrateComments &&
        celebrateComments.comments.findIndex(c => c.id === editingCommentId);
      if (index && index >= 0) {
        listRef.current &&
          listRef.current.scrollToIndex({ index, viewPosition: 1 });
        return;
      }
    }
    scrollToEnd();
  };

  useKeyboardListeners({ onShow: () => scrollToFocusedRef() });

  const refreshComments = () => {
    return dispatch(reloadCelebrateComments(event.id, organization.id));
  };

  const { isRefreshing, refresh } = useRefreshing(refreshComments);

  const renderHeader = () => (
    <SafeAreaView>
      <StatusBar {...theme.statusBar.darkContent} />
      <View style={styles.header}>
        <View style={{ flexDirection: 'row' }}>
          <View style={{ flex: 1 }}>
            <CommunityFeedItemName
              name={event.subjectPersonName}
              person={event.subjectPerson}
              communityId={organization.id}
              pressable={true}
            />
            <CardTime date={event.createdAt} />
          </View>
          {!eventIsNotLoaded && (
            <CommentLikeComponent
              item={event}
              communityId={organization.id}
              onRefresh={onRefreshCelebrateItem}
            />
          )}
          <DeprecatedBackButton
            style={styles.backButtonStyle}
            iconStyle={styles.backButtonIconStyle}
            customIcon="deleteIcon"
          />
        </View>
      </View>
    </SafeAreaView>
  );

  const renderCommentsList = () => (
    <View style={styles.contentContainer}>
      <Image source={TRAILS1} style={styles.trailsTop} />
      <Image source={TRAILS2} style={styles.trailsBottom} />
      {!eventIsNotLoaded && (
        <CommentsList
          //@ts-ignore
          event={event} //TODO: Modify to use CommunityFeedItem type
          organization={organization}
          listProps={{
            ref: listRef,
            refreshControl: (
              <RefreshControl
                testID="RefreshControl"
                refreshing={isRefreshing}
                onRefresh={refresh}
              />
            ),
            ListHeaderComponent: () => (
              <CommunityFeedItemContent
                item={event}
                communityId={organization.id}
                style={styles.itemContent}
              />
            ),
          }}
        />
      )}
    </View>
  );

  const renderCommentBox = () => (
    <CelebrateCommentBox
      //@ts-ignore
      event={event} //TODO: Modify to use CommunityFeedItem type
      onAddComplete={scrollToEnd}
      organization={organization}
    />
  );

  return (
    <View style={styles.pageContainer}>
      {renderHeader()}
      {renderCommentsList()}
      {renderCommentBox()}
    </View>
  );
};

const mapStateToProps = (
  {
    auth,
    organizations,
    celebrateComments,
  }: {
    auth: AuthState;
    organizations: OrganizationsState;
    celebrateComments: CelebrateCommentsState;
  },
  {
    navigation: {
      state: {
        params: { item, orgId },
      },
    },
  }: // eslint-disable-next-line @typescript-eslint/no-explicit-any
  any,
) => {
  const { subjectPerson } = item as CommunityFeedItem;
  const organization = organizationSelector({ organizations }, { orgId });

  return {
    organization,
    celebrateComments: celebrateCommentsSelector(
      { celebrateComments },
      { eventId: item.id },
    ),
    editingCommentId: celebrateComments.editingCommentId,
    analyticsAssignmentType: getAnalyticsAssignmentType(
      subjectPerson,
      auth,
      organization,
    ),
    analyticsPermissionType: getAnalyticsPermissionType(auth, organization),
  };
};

export default connect(mapStateToProps)(CelebrateDetailScreen);
export const CELEBRATE_DETAIL_SCREEN = 'nav/CELEBRATE_DETAIL_SCREEN';
