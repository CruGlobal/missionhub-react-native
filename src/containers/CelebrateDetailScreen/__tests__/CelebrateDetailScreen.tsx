import 'react-native';
import React from 'react';
import MockDate from 'mockdate';
import { fireEvent } from 'react-native-testing-library';

import { renderWithContext } from '../../../../testUtils';
import { mockFragment } from '../../../../testUtils/apolloMockClient';
import { organizationSelector } from '../../../selectors/organizations';
import { useKeyboardListeners } from '../../../utils/hooks/useKeyboardListeners';
import {
  reloadCelebrateComments,
  resetCelebrateEditingComment,
} from '../../../actions/celebrateComments';
import CommentsList from '../../CommentsList';
import { celebrateCommentsSelector } from '../../../selectors/celebrateComments';
import { Organization } from '../../../reducers/organizations';
import { CelebrateComment } from '../../../reducers/celebrateComments';
import { CELEBRATE_ITEM_FRAGMENT } from '../../../components/CelebrateItem/queries';
import { GetCelebrateFeed_community_celebrationItems_nodes as CelebrateItem } from '../../CelebrateFeed/__generated__/GetCelebrateFeed';

import CelebrateDetailScreen from '..';

jest.mock('../../../utils/hooks/useKeyboardListeners');
jest.mock('../../../selectors/celebration');
jest.mock('../../../selectors/celebrateComments');
jest.mock('../../../selectors/organizations');
jest.mock('../../../actions/celebrateComments');
jest.mock('../../Analytics', () => 'Analytics');
jest.mock('../../CommentItem', () => 'CommentItem');
jest.useFakeTimers();

MockDate.set('2019-04-12 12:00:00', 300);

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const comments: { comments: CelebrateComment[]; pagination: any } = {
  comments: [
    {
      id: 'comment1',
      person: { first_name: 'Person', last_name: '1' },
      content: 'some comment',
      created_at: '2019-04-11T13:51:49.888',
      updated_at: '2019-04-11T13:51:49.888',
    },
    {
      id: 'comment2',
      person: { first_name: 'Person', last_name: '2' },
      content: 'some comment',
      created_at: '2019-04-11T13:51:49.888',
      updated_at: '2019-04-11T13:51:49.888',
    },
  ],
  pagination: {},
};

const myId = 'myId';
const orgId = '24234234';
const organization: Organization = { id: orgId, name: 'Community' };
const event = mockFragment<CelebrateItem>(CELEBRATE_ITEM_FRAGMENT);
const organizations = [organization];
const celebrateComments = { editingCommentId: null };
const auth = { person: { id: myId } };

const initialState = {
  organizations,
  celebrateComments,
  auth,
};

let onShowKeyboard: () => void;

beforeEach(() => {
  (useKeyboardListeners as jest.Mock).mockImplementation(
    ({ onShow }: { onShow: () => void }) => (onShowKeyboard = onShow),
  );
  (reloadCelebrateComments as jest.Mock).mockReturnValue({
    type: 'reloadCelebrateComments',
  });
  (resetCelebrateEditingComment as jest.Mock).mockReturnValue({
    type: 'resetCelebrateEditingComment',
  });
  ((organizationSelector as unknown) as jest.Mock).mockReturnValue(
    organization,
  );
  ((celebrateCommentsSelector as unknown) as jest.Mock).mockReturnValue(
    comments,
  );
});

it('renders correctly', () => {
  renderWithContext(<CelebrateDetailScreen />, {
    initialState,
    navParams: { event, orgId },
  }).snapshot();

  expect(organizationSelector).toHaveBeenCalledWith(
    { organizations },
    { orgId },
  );
  expect(celebrateCommentsSelector).toHaveBeenCalledWith(
    {
      celebrateComments,
    },
    { eventId: event.id },
  );
});

describe('refresh', () => {
  it('calls refreshComments', () => {
    const { getByTestId } = renderWithContext(<CelebrateDetailScreen />, {
      initialState,
      navParams: { event, orgId },
    });

    fireEvent(getByTestId('RefreshControl'), 'onRefresh');

    expect(reloadCelebrateComments).toHaveBeenCalledWith(
      event.id,
      organization.id,
    );
  });
});

describe('celebrate add complete', () => {
  it('scrolls to end on add complete', () => {
    const scrollToEnd = jest.fn();

    const { getByType, getByTestId } = renderWithContext(
      <CelebrateDetailScreen />,
      {
        initialState,
        navParams: { event, orgId },
      },
    );

    getByType(
      CommentsList,
    ).props.listProps.ref.current.scrollToEnd = scrollToEnd;

    fireEvent(getByTestId('CelebrateCommentBox'), 'onAddComplete');

    expect(scrollToEnd).toHaveBeenCalledWith();
  });
});

describe('keyboard show', () => {
  it('without editing comment', () => {
    const scrollToEnd = jest.fn();

    const { getByType } = renderWithContext(<CelebrateDetailScreen />, {
      initialState,
      navParams: { event, orgId },
    });

    getByType(
      CommentsList,
    ).props.listProps.ref.current.scrollToEnd = scrollToEnd;

    onShowKeyboard();

    expect(scrollToEnd).toHaveBeenCalledWith();
  });

  it('with editing comment', () => {
    const scrollToIndex = jest.fn();

    const { getByType } = renderWithContext(<CelebrateDetailScreen />, {
      initialState: {
        ...initialState,
        celebrateComments: { editingCommentId: comments.comments[1].id },
      },
      navParams: { event, orgId },
    });

    getByType(
      CommentsList,
    ).props.listProps.ref.current.scrollToIndex = scrollToIndex;

    onShowKeyboard();

    expect(scrollToIndex).toHaveBeenCalledWith({
      index: 1,
      viewPosition: 1,
    });
  });

  it('with editing comment that doesnt exist', () => {
    const scrollToEnd = jest.fn();

    const { getByType } = renderWithContext(<CelebrateDetailScreen />, {
      initialState: {
        ...initialState,
        celebrateComments: { editingCommentId: 'doesnt exist' },
      },
      navParams: { event, orgId },
    });

    getByType(
      CommentsList,
    ).props.listProps.ref.current.scrollToEnd = scrollToEnd;

    onShowKeyboard();

    expect(scrollToEnd).toHaveBeenCalledWith();
  });
});
