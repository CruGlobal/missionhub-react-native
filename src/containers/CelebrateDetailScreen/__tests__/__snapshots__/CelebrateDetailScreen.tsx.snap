// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Analytics
    screenName={
      Array [
        "celebrate item",
        "comments",
      ]
    }
  />
  <SafeAreaView>
    <StatusBar
      animated={true}
      backgroundColor="#006789"
      barStyle="dark-content"
      showHideTransition="fade"
    />
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "padding": 14,
          "paddingBottom": 5,
        }
      }
    >
      // @ts-ignore
      <View
        flexDirection="row"
      >
        // @ts-ignore
        <View
          flex={1}
        >
          <Connect(withI18nextTranslation(CelebrateItemName))
            name="Roger"
            organization={
              Object {
                "id": "24234234",
              }
            }
            pressable={true}
          />
          <CardTime
            date="2019-04-11T13:51:49.888"
          />
        </View>
        // @ts-ignore
        <Connect(CommentLikeComponent)
          event={
            Object {
              "changed_attribute_value": "2019-04-11T13:51:49.888",
              "id": "90001",
              "organization": Object {
                "id": "24234234",
              },
              "subject_person_name": "Roger",
            }
          }
        />
        <Connect(BackButton)
          customIcon="deleteIcon"
          iconStyle={
            Object {
              "color": "#000000",
            }
          }
          style={
            Object {
              "marginLeft": 10,
            }
          }
        />
      </View>
    </View>
  </SafeAreaView>
  <View
    style={
      Object {
        "backgroundColor": "#505256",
        "flex": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/Trailss.png",
        }
      }
      style={
        Object {
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/TrailGrey.png",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
        }
      }
    />
    <Connect(withI18nextTranslation(CommentsList))
      event={
        Object {
          "changed_attribute_value": "2019-04-11T13:51:49.888",
          "id": "90001",
          "organization": Object {
            "id": "24234234",
          },
          "subject_person_name": "Roger",
        }
      }
      listProps={
        Object {
          "ListHeaderComponent": [Function],
          "ref": [Function],
          "refreshControl": <MyRefreshControl
            onRefresh={[Function]}
            refreshing={false}
          />,
        }
      }
    />
  </View>
  <Connect(CelebrateCommentBox)
    event={
      Object {
        "changed_attribute_value": "2019-04-11T13:51:49.888",
        "id": "90001",
        "organization": Object {
          "id": "24234234",
        },
        "subject_person_name": "Roger",
      }
    }
    onAddComplete={[Function]}
  />
</View>
`;
